taxcalc/taxbrain/consumption.py

```diff
@@ -0,0 +1,133 @@
+                                          wrt_full_compensation=False)
+    # compare unweighted mtr results with and without consumption response
+    epsilon = 1.0e-6  # this would represent a mtr of 0.0001 percent
+    assert np.allclose(mtr1_fica, mtr0_fica, atol=epsilon, rtol=0.0)
+    mtr_raw_diff = mtr1_itax - mtr0_itax
+    mtr1_itax = np.where(np.logical_and(mtr_raw_diff > 0.0,
+                                        mtr_raw_diff < epsilon),
+                         mtr0_itax, mtr1_itax)  # zero out small positive diffs
+    num_total = mtr1_itax.size
+    num_increases = np.sum(np.greater(mtr1_itax, mtr0_itax))
+    num_decreases = np.sum(np.less(mtr1_itax, mtr0_itax))
+    num_nochanges = num_total - num_increases - num_decreases
+    res = 'unweighted_num_of_mtr_{}_with_consump_response= {:6d} ({:5.1f}%)\n'
+    sys.stdout.write(res.format('increases', num_increases,
+                                (100.0 * num_increases) / num_total))
+    sys.stdout.write(res.format('decreases', num_decreases,
+                                (100.0 * num_decreases) / num_total))
+    sys.stdout.write(res.format('nochanges', num_nochanges,
+                                (100.0 * num_nochanges) / num_total))
+    sys.stdout.write(res.format('all_units', num_total, 100.0))
+    # compute average size of decreases in mtr_itax
+    mtr_pre = mtr0_itax[mtr1_itax < mtr0_itax]
+    assert mtr_pre.size == num_decreases
+    avg_pre = np.mean(mtr_pre)
+    res = 'unweighted_abs_mean_no_c_resp(for_decreases)mtr_itax= {:.4f}\n'
+    sys.stdout.write(res.format(avg_pre))
+    mtr_diff = mtr1_itax - mtr0_itax
+    mtr_neg = mtr_diff[mtr_diff < 0.0]
+    assert mtr_neg.size == num_decreases
+    avg_neg = np.mean(mtr_neg)
+    assert avg_neg < 0.0
+    res = 'unweighted_abs_mean_change(for_decreases)in_mtr_itax= {:.4f}\n'
+    sys.stdout.write(res.format(-avg_neg))
+    res = '   ratio_of_abs_change_in_mtr_itax_and_no_c_resp_mtr_itax= {:.3f}\n'
+    sys.stdout.write(res.format(-avg_neg / avg_pre))
+    # compare weighted mtr results with and without consumption response
+    wghts_pre = wghts[mtr1_itax < mtr0_itax]
+    assert wghts_pre.size == num_decreases
+    res = 'weighted_percent_of_units_with_mtr_decrease= {:.1f}%\n'
+    frac = wghts_pre.sum() / wghts.sum()
+    sys.stdout.write(res.format(100.0 * frac))
+    res = '   ratio_of_abs_change_in_mtr_itax_and_no_c_resp_mtr_itax= {:.3f}\n'
+    w_avg_pre = np.mean((mtr_pre * wghts_pre).sum() / wghts_pre.sum())
+    w_avg_neg = np.mean((mtr_neg * wghts_pre).sum() / wghts_pre.sum())
+    sys.stdout.write(res.format(-w_avg_neg / w_avg_pre))
+    # return no-error exit code
+    return 0
+# end of main function code
+
+
+if __name__ == '__main__':
+    # parse command-line arguments:
+    PARSER = argparse.ArgumentParser(
+        prog='python consumption.py',
+        formatter_class=argparse.RawTextHelpFormatter,
+        description=('Writes to stdout marginal-tax-rate results --- both '
```

Comment:
In the future, you could use triply quoted strings here for long text passages that respect new-line characters. That way you would not have to put the `\n` in manually.